includes/class-amp-rest-api.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * AMP REST API.
+ *
+ * @package AMP
+ * @since 2.0
+ */
+
+/**
+ * Class AMP_REST_API.
+ */
+class AMP_REST_API
+{
+
+  /**
+   * Init.
+   */
+  public static function init()
+  {
+    if (!class_exists('AMP_REST_API')) {
+      return;
+    }
+
+    // Register one rest_prepare filter for each custom post type.
+    add_action('registered_post_type', array(__CLASS__, 'add_rest_prepare_filter'));
```

Comment:
This action runs very early. I think this should be deferred until the REST API is initialized.

Also, we only want to add this to post types that support AMP (as it can be disabled under some circumstances).

Perhaps like this:

```php

add_action( 'rest_api_init', [ __CLASS__, 'rest_api_init' ] );

// further down ...

public static function rest_api_init() {
	foreach( AMP_Post_Type_Support::get_builtin_supported_post_types() as $post_type ) {
		$post_type_supported = post_type_supports( $post_type, AMP_Post_Type_Support::SLUG );

		if ( $post_type_supported ) {
			add_filter( 'rest_prepare_' . $post_type, [ __CLASS__, 'add_content_amp_field' ], 10, 3 );
		}
	}
}
```

Reply:
Ok, I've used `amp_init` to load the class, switched to `rest_api_init` and used only post types from `get_builtin_supported_post_types`.

All that is working great 👍 

Oh, and I have prettified it properly this time (I hope).